index.bs

```diff
@@ -2227,6 +2227,46 @@ During the above process, the user agent SHOULD show some UI to the user to guid
 authorizing an authenticator. When <code>|options|.{{CredentialCreationOptions/mediation}}</code> is set to {{CredentialMediationRequirement/conditional}}, prominent modal UI should <i>not</i> be shown <i>unless</i> credential creation was previously consented to via means determined by the user agent.
 </div>
 
+#### Registration API Exceptions #### {#sctn-registration-api-exceptions}
+
+[=[WRPS]=] can encounter the following {{DOMException|DOMExceptions}} from a call to {{CredentialsContainer/create()|navigator.credentials.create()}}. Some errors can have multiple reasons for why they happened, requiring the [=[WRPS]=] to infer the actual reason based on their use of WebAuthn:
+
+<dl>
+    :   {{AbortError}}
+    ::  The ceremony was cancelled by an {{AbortController}}.
+        See [[#sctn-abortoperation]] and [[#sctn-sample-aborting]].
+
+    :   {{ConstraintError}}
+    ::  Either {{residentKey}} was set to {{ResidentKeyRequirement/required}} and no available authenticator supported resident keys,
+        or {{AuthenticatorSelectionCriteria/userVerification}} was set to {{UserVerificationRequirement/required}} and no available authenticator could perform [=user verification=].
+
+    :   {{InvalidStateError}}
+    ::  The authenticator used in the ceremony recognized an entry in {{PublicKeyCredentialCreationOptions/excludeCredentials}}
+        after the user [=user consent|consented=] to registering a credential.
+
+    :   {{NotSupportedError}}
+    ::  No entry in {{PublicKeyCredentialCreationOptions/pubKeyCredParams}} had a {{PublicKeyCredentialDescriptor/type}} property of {{PublicKeyCredentialType/public-key}},
+        or the [=authenticator=] did not support any of the specified cryptographic parameters in {{PublicKeyCredentialCreationOptions/pubKeyCredParams}}.
+
+    :   {{SecurityError}}
+    ::  The [=effective domain=] was not a [=valid domain=],
+        or <code>{{PublicKeyCredentialCreationOptions/rp}}.{{PublicKeyCredentialRpEntity/id}}</code> was not a registrable domain suffix of nor was equal to the [=effective domain=].
+
+    :   {{TypeError}}
+    ::  The <code>|options|</code> argument was not a valid [[#sctn-credentialcreationoptions-extension|CredentialCreationOptions]] value,
+        or the value of <code>{{PublicKeyCredentialCreationOptions/user}}.{{PublicKeyCredentialUserEntity/id}}</code> was empty or was longer than 64 bytes.
+
+    :   {{UnknownError}}
+    ::  The [=authenticator=] could not process the supplied options,
+        or encountered an error while creating the new credential.
+
+    :   {{NotAllowedError}}
+    ::  A catch-all error covering a wide range of possible reasons,
+        including common ones like the user canceling out of the ceremony.
+        Some of these causes are documented throughout this spec,
+        while others are client-specific.
+
```

Comment:
Perhaps we should also point out that extensions may also introduce new error cases which are not listed here (because extensions may be defined outside the WebAuthn spec).

Reply:
I tweaked both of these `NotAllowedError` descriptions with text to the following effect:
![Screenshot 2024-07-09 at 3 15 00 PM](https://github.com/w3c/webauthn/assets/5166470/d2b585fd-6460-4618-84f0-5081faf3c77a)
Is this what you had in mind?

Reply:
I didn't mean `NotAllowedError` specifically, rather that extensions may introduce additional error cases in genera. For example, the [PRF extension](https://w3c.github.io/webauthn/#prf-extension) returns `NotSupportedError` and `SyntaxError` in some circumstances.

Reply:
Thanks for clarifying, I took another stab at it, moving the mention of the possiblity of unlisted extension exceptions up into the intro:

![Screenshot 2024-07-17 at 9 31 55 AM](https://github.com/user-attachments/assets/5a1fdc77-dccd-4e71-8024-0256b012493f)

Reply:
Looks good! But now that implies that errors from extensions defined within the spec _will_ be listed here. Should we do that? :slightly_smiling_face:

Reply:
Alright, point taken. I've decided to punt on capturing extension exceptions here as, after reviewing a few extensions, I think it'd make these new sections too noisy. I'd like to keep this to exceptions that can occur from typical calls to `.create()` and `.get()`, which I'm going to assume don't include any use of extensions.

Reply:
@emlun Here's a preview of what I have so far:

![Screenshot 2024-07-23 at 2 50 02 PM](https://github.com/user-attachments/assets/b790b671-78ce-4516-95c9-f72b70df8ba2)

Reply:
Looks good!